doc/whats_new/v1.6.rst

```diff
@@ -277,9 +277,13 @@ Changelog
 
 - |Fix| :class:`linear_model.LassoCV` and :class:`linear_model.ElasticNetCV` now
   take sample weights into accounts to define the search grid for the internally tuned
-  `alpha` hyper-parameter. :pr:`29442` by :user:`John Hopfensperger <s-banach> and
+  `alpha` hyper-parameter. :pr:`29442` by :user:`John Hopfensperger <s-banach>` and
   :user:`Shruti Nath <snath-xoc>`.
 
+- |Fix| The Hessian warning in :class:`linear_model._linear_loss.LinearModelLoss` is now
+  sample weight aware.
```

Comment:
We should only mention public API in the changelog entries. Users are not necessarily aware about which estimator is impacted by this change.

- `linear_model.LogisticRegression`, `linear_model.PoissonRegressor`, `linear_model.GammaRegressor`, `linear_model.TweedieRegressor` now take sample weight into account to decide on when to fall back to `solver="lbfgs"` whenever `solver="newton-cholesky"` becomes numerically unstable.